lib/combine_pdf/decrypt.rb

```diff
@@ -143,12 +140,16 @@ def decrypt_AES(encrypted, encrypted_id, encrypted_generation, _encrypted_filter
       object_key << 'sAlT'.force_encoding(Encoding::ASCII_8BIT)
       key_length = object_key.length < 16 ? object_key.length : 16
 
-      cipher = OpenSSL::Cipher.new("aes-#{key_length << 3}-cbc")
-      cipher.decrypt
-      cipher.key = Digest::MD5.digest(object_key)[(0...key_length)]
-      cipher.iv = encrypted[0..15]
-      cipher.padding = 0
-      cipher.update(encrypted[16..-1]) + cipher.final
+      begin
+        cipher = OpenSSL::Cipher.new("aes-#{key_length << 3}-cbc")
+        cipher.decrypt
+        cipher.key = Digest::MD5.digest(object_key)[(0...key_length)]
+        cipher.iv = encrypted[0..15]
+        cipher.padding = 0
+        cipher.update(encrypted[16..-1]) + cipher.final
+      rescue Exception => e
```

Comment:
Thanks for telling me.

I didn't solve this issue, since I noticed that letting failures pass through was actually working. It was allowing non-encrypted data to pass through and throwing exceptions when trying to deflate encrypted data that wasn't decrypted properly.

I'll have to look into this and see why non-encrypted data is being sent to the decryption method and how to address this, but this doesn't seem to effect the decryption process for now.

I don't know if I mentioned this, but I really enjoyed working with you on this. Your changes were minimal, non-intrusive and very effective.

Thank you.